[2024-07-17T10:10:19.359+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-07-17T10:10:19.426+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: computer_data.insert_battery_data_task scheduled__2024-07-17T10:06:00+00:00 [queued]>
[2024-07-17T10:10:19.442+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: computer_data.insert_battery_data_task scheduled__2024-07-17T10:06:00+00:00 [queued]>
[2024-07-17T10:10:19.443+0000] {taskinstance.py:2306} INFO - Starting attempt 2 of 2
[2024-07-17T10:10:19.481+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): insert_battery_data_task> on 2024-07-17 10:06:00+00:00
[2024-07-17T10:10:19.587+0000] {warnings.py:110} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61: DeprecationWarning: This process (pid=193) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-07-17T10:10:19.591+0000] {standard_task_runner.py:63} INFO - Started process 195 to run task
[2024-07-17T10:10:19.592+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'computer_data', 'insert_battery_data_task', 'scheduled__2024-07-17T10:06:00+00:00', '--job-id', '18', '--raw', '--subdir', 'DAGS_FOLDER/collect_to_postgre_dag.py', '--cfg-path', '/tmp/tmp4xx2xrg0']
[2024-07-17T10:10:19.618+0000] {standard_task_runner.py:91} INFO - Job 18: Subtask insert_battery_data_task
[2024-07-17T10:10:19.725+0000] {task_command.py:426} INFO - Running <TaskInstance: computer_data.insert_battery_data_task scheduled__2024-07-17T10:06:00+00:00 [running]> on host be9d87096f50
[2024-07-17T10:10:19.933+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='computer_data' AIRFLOW_CTX_TASK_ID='insert_battery_data_task' AIRFLOW_CTX_EXECUTION_DATE='2024-07-17T10:06:00+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-07-17T10:06:00+00:00'
[2024-07-17T10:10:19.967+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-07-17T10:10:19.970+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-07-17T10:10:19.973+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 400, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/battery.py", line 17, in insert_battery_data
    collect_data = get_battery()
                   ^^^^^^^^^^^^^
  File "/opt/airflow/dags/battery.py", line 10, in get_battery
    battery_info_list.append(battery_info.percent)
                             ^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'percent'
[2024-07-17T10:10:20.022+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=computer_data, task_id=insert_battery_data_task, run_id=scheduled__2024-07-17T10:06:00+00:00, execution_date=20240717T100600, start_date=20240717T101019, end_date=20240717T101020
[2024-07-17T10:10:20.052+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 18 for task insert_battery_data_task ('NoneType' object has no attribute 'percent'; 195)
[2024-07-17T10:10:20.100+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-07-17T10:10:20.117+0000] {taskinstance.py:3498} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-07-17T10:10:20.119+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
